### DISCUSSION TITLE
How to use the !include instruction

### DISCUSSION THREAD

**Discussion by pvyleta** [john30/ebusd#1216] [Q&A]
I recently bought a heat recovery unit (HRU) from Brink which has eBus, and I started playing around with it. Turns out most HRU from Brink have similar parameters, so I thought it would be nice to create a nice structure, similar to that Vaillant has in [ebusd-configuration](https://github.com/john30/ebusd-configuration/tree/master/ebusd-2.1.x/en/vaillant).

However, I am struggling to understand what should I fill in the `.inc` file to make it work. Here is my file `3c.csv`
```
# type (r[1-9];w;u),circuit,name,[comment],[QQ],ZZ,PBSB,[ID],field1,part (m/s),datatypes/templates,divider/values,unit,comment,field2,part (m/s),datatypes/templates ,divider/values,unit,comment,field3,part (m/s),datatypes/templates,divider/values,unit,comment,field4,part (m/s),datatypes/templates,divider/values,unit,comment,field5,part (m/s),datatypes/templates,divider/values,unit,comment
*r,sky300,,,,3c,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
r,sky300,FanMode,,,,4022,01,,,UIR,0=Min;1=Low;2=Medium;3=High,,FanMode
!include,brink.inc,,,,,,,,,,,,
```

and here is my `brink.inc` file
```
# type (r[1-9];w;u),circuit,name,[comment],[QQ],ZZ,PBSB,[ID],field1,part (m/s),datatypes/templates,divider/values,unit,comment,field2,part (m/s),datatypes/templates,divider/values,unit,comment,field3,part (m/s),datatypes/templates,divider/values,unit,comment,field4,part (m/s),datatypes/templates,divider/values,unit,comment,field5,part (m/s),datatypes/templates,divider/values,unit,comment
r,,FlowMode0,FlowMode0,,,4050,21,,,SIR,,m³/h,,Min,,SIR,,m³/h,[min:0],Max,,SIR,,m³/h,[max:50],Step,,SIR,,m³/h,[step:50],Default,,SIR,,m³/h,[default:50]

```
Both files live in the same folder. When I try to test my config via `ebusd --checkconfig -c /config/ebusd-configuration/latest/en` I receive the following error:

`2024-03-05 17:07:55.016 [main error] error executing instructions: ERR: missing argument, last error: error including "encon/brink.inc" for "3c": ERR: missing argument encon/brink.inc:2: ERR: missing argument, circuit`

Is anyone able to spot what am I doing wrong? I thought that having the line `*r,sky300,,,,3c,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,` in my `3c.csv` would apply to the include as well. Is there a way for me to not have to specify the circuit in the included file explicitly?

**Comment by rbaleja** [comment]
I'm also starting my adventure with ebus, but it seems to me that you don't have to use include at the beginning. First, create one configuration file and if it works, then transfer the features that are common to your configuration to *.inc files. the format of these files is the same as *.csv files

I don't know if this command is too long, but let the experienced people comment on it
